lib/gui/app.js

```diff
@@ -162,7 +175,9 @@ app.run(($timeout, DriveScannerService, DrivesModel, ErrorService) => {
     // available drives list has changed, and incorrectly
     // keeps asking the user to "Connect a drive".
     $timeout(() => {
-      DrivesModel.setDrives(drives);
+      if (!_.isEqual(DrivesModel.getDrives(), drives)) {
+        DrivesModel.setDrives(drives);
```

Comment:
Even with the custom comparison functions you needed to add? Or am I getting my wires crossed? (more than likely!)